src/main/java/org/opensearch/neuralsearch/processor/ScoreNormalizationTechnique.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.neuralsearch.processor;
+
+import lombok.Builder;
+import lombok.Data;
+
+import com.google.common.primitives.Floats;
+
+/**
+ * Collection of techniques for score normalization
+ */
+public enum ScoreNormalizationTechnique {
+
+    /**
+     * Min-max normalization method.
+     * nscore = (score - min_score)/(max_score - min_score)
+     */
+    MIN_MAX {
+        @Override
+        float normalize(ScoreNormalizationRequest request) {
+            // edge case when there is only one score and min and max scores are same
+            if (Floats.compare(request.getMaxScore(), request.getMinScore()) == 0
+                && Floats.compare(request.getMaxScore(), request.getScore()) == 0) {
+                return SINGLE_RESULT_SCORE;
+            }
```

Comment:
This is really interesting use case, but I am not sure making this as 0.01f will work? reason being if that result was the most relevant result, then it will never be returned to the customer. can we check with Science team on this what is their thought.

Reply:
I'm actually making it 1.0, max possible in normalization.